core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -425,6 +445,9 @@ UNWIND(	.cantunwind)
 
 	bl	plat_cpu_reset_late
 
+    /* this core just comes out of reset, discard undefined l1 contents */
+	bl	arm_cl1_d_invbysetway
```

Comment:
This is safe because its "cache invalidate" on L1, other cores would not respond to this op. Besides, the MMU and cache for secondary core is not enabled yet at this moment, so I believe there will not be any coherency broadcast even if the SMP bit is enable.

Anyway, my testing on i.MX6 (Cortex-A9) shows this operation is required for secondary cores to boot up. ARM specs also mentioned that when CPU comes out of reset, the cache content is UNDEFINED, so an all set-way invalidation is necessary.

And I wonder why this is not required before. Maybe this again is because the OP-TEE is targeted for Firmware; so it is possible that Firmware cleaned the cache before putting secondary cores into OP-TEE; however, for VxWorks's usage, the secondary core is released from reset and is directly put into OP-TEE, so the cache should be handled somehow here.

Anyway, I admit globally changing this is inconsiderate, I'll put it inside CFG_TEE_VXWORKS_SUPPORT to reflect this is due to VxWorks's special boot scenario.